core/matrix/batch_struct.hpp

```diff
@@ -82,6 +83,53 @@ struct uniform_batch {
 }  // namespace batch_dense
 
 
+namespace batch_ell {
+
+
+/**
+ * Encapsulates one matrix from a batch of ell matrices.
+ */
+template <typename ValueType>
+struct batch_item {
+    using value_type = ValueType;
+    using index_type = int32;
+
+    ValueType* values;
+    const index_type* col_idxs;
+    index_type stride;
+    index_type num_rows;
+    index_type num_cols;
+    index_type num_stored_elems_per_row;
+};
+
+
+/**
+ * A 'simple' structure to store a global uniform batch of ell matrices.
+ */
+template <typename ValueType>
+struct uniform_batch {
+    using value_type = ValueType;
+    using index_type = int;
```

Comment:
This explicitly restricts the kernel level representation of ELL to use int32. I think we need to make both representations consistent, so either add a template here, or remove the index template from Ell.